src/main/java/seedu/address/model/house/UniqueHouseList.java

```diff
@@ -33,24 +33,20 @@ public class UniqueHouseList implements Iterable<House> {
      */
     public boolean contains(House toCheck) {
         requireNonNull(toCheck);
-        if (toCheck instanceof Landed) {
-            return internalList.stream().anyMatch(toCheck::equals);
-        } else {
-            for (House h: internalList) {
-                int index1 = toCheck.toString().indexOf("Street");
-                int index2 = h.toString().indexOf("Street");
-
-                assert index1 != -1 : "Invalid house";
-                assert index2 != -1 : "Invalid house";
-
-                String substr1 = toCheck.toString().substring(index1);
-                String substr2 = h.toString().substring(index2);
-                if (substr1.equals(substr2)) {
-                    return true;
-                }
+        for (House h: internalList) {
+            int index1 = toCheck.toString().toLowerCase().indexOf("street");
```

Comment:
shall change the "index1" & "index2" etc variable names to a more meaningful name.